engine/src/main/java/org/hibernate/search/query/dsl/sort/SortDistanceContext.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.query.dsl.sort;
+
+import org.hibernate.search.query.dsl.Unit;
+
+/**
+ * @author Emmanuel Bernard emmanuel@hibernate.org
+ */
+public interface SortDistanceContext extends SortOrder<SortDistanceContext>, SortAdditionalSortFieldContext, SortTermination {
+
+	/**
+	 * Hint for the backend to return data in the appropriate unit.
+	 */
+	SortDistanceContext in(Unit unit);
```

Comment:
@Sanne when you do a sort by distance, ES includes the distance calculated in the results so that you don't have to calculate it yourself. This is why they expose a way to tweak the unit. We use it in our ES work to avoid using a script to calculate the distance.

See https://github.com/hibernate/hibernate-search/blob/master/elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchHSQueryImpl.java#L541 .